starter_code/src/movies.js

```diff
@@ -1,15 +1,166 @@
 /* eslint no-restricted-globals: 'off' */
 
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+//..sort by year
+function orderByYear(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array))
+    return orderByTitle(array).sort((a, b) => a["year"] - b["year"]);
+}
+
+//..sort by title (only if same year)
+function orderByTitle(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array))
+    return newMoviesArray.sort(function(a, b) {
+      if (a["title"] < b["title"]) {
+        return -1;
+      } 
+      if (a["title"] > b["title"]) {
+        return 1;
+      }
+      if (a["title"] == b["title"]) {
+        return 0;
+      }
+    }); 
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+function theSpielbergMovies(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array));
+    const spielbergMovies = newMoviesArray.filter(theDirector => theDirector["director"] == "Steven Spielberg");
+    return spielbergMovies;
+}
+  
+function theDramaMovies(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array));
+    const theDramaMovies = newMoviesArray.filter(theDrama => theDrama["genre"].includes("Drama"));
+    return theDramaMovies;
+}
+
+function howManyMovies(array) {
+    const dramaArray = theDramaMovies(array);
+    return theSpielbergMovies(dramaArray).length;
+}
 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+//..cut down to 20
+function cutArrayToTwenty(array) {
+  const newMoviesArray = JSON.parse(JSON.stringify(array))
+  if (newMoviesArray.length > 20) {
+    return newMoviesArray.slice(0,20);
+  } else {
+    return newMoviesArray;
+  }
+}
+
+//..Get titles only
+function getTitleOnly(array) {
+  const newMoviesArray = JSON.parse(JSON.stringify(array));
+  const titlesArray = newMoviesArray.map(theTitle => {
+    return theTitle["title"]
+  })
+  return titlesArray
+}
+
+function orderAlphabetically(array) {
+  let twentyMovies = orderByTitle(array);
+  twentyMovies = cutArrayToTwenty(twentyMovies);
+  twentyMovies = getTitleOnly(twentyMovies);
+  return twentyMovies;
+}
 
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+//..filter objects with score and keep socre only in new array
+function theAverageScoreArray(array) {
+  const newMoviesArray = JSON.parse(JSON.stringify(array));
+  const theAverageScore = newMoviesArray.filter(callback => callback["rate"] > 0);
+  const allRates = theAverageScore.map(theRate => theRate["rate"]);
+  return allRates;
+}
+
+//.. calculate average rate
+function ratesAverage(array) {
+  if (array.length < 1 || array.length == null || array.length == undefined) {
+    return 0
+  } else {
+    return +(theAverageScoreArray(array).reduce((accumulator, currentValue) => accumulator + currentValue)/array.length).toFixed(2);
+  }
+}
+
```

Comment:
Nice. You don't have to use the ternary operator but I just want to mention how you could use it for this function.

function ratesAverage(array) {
  return array.length < 1 || array.length == null || array.length == undefined ? 0 :
  +(theAverageScoreArray(array).reduce((accumulator, currentValue) => accumulator + currentValue)/array.length).toFixed(2);
}
console.log(ratesAverage(movies))